src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -811,6 +811,13 @@ XBUtilities::string_to_bytes(std::string str)
   return size;
 }
 
+bool 
+XBUtilities::
+isPowerOf2(const uint64_t x)
+{
+  return !(x & (x - 1));
```

Comment:
There is a bug here.  If x is zero it returns back an incorrect result for zero isn't a power of two.
```suggestion
  return !(x ==0) && !(x & (x - 1));
```